src/Store.js

```diff
@@ -0,0 +1,50 @@
+import invariant from 'invariant';
+import isPlainObject from './utils/isPlainObject';
+
+export default class Store {
+  constructor(reducer, initialState) {
+    invariant(
+      typeof reducer === 'function',
+      'Expected the reducer to be a function.'
+    );
+
+    this.state = initialState;
+    this.listeners = [];
+    this.replaceReducer(reducer);
+  }
+
+  getReducer() {
+    return this.reducer;
+  }
+
+  replaceReducer(nextReducer) {
+    this.reducer = nextReducer;
+    this.dispatch({ type: '@@INIT' });
+  }
+
+  dispatch(action) {
```

Comment:
I agree in a way. I'm not sure it's worth changing though. It doesn't seem to cause much confusion (unlike “Stores” naming which did, so we're changing it to “Reducers”).